app/models/user.rb

```diff
@@ -54,22 +54,53 @@ def administrated_projects
     scope
   end
 
-  def self.to_csv
-    @users = User.order(:id)
+  def self.to_csv(
+    inherited: false, deleted: false, project_id: nil, user_id: nil,
+    datetime: (Time.now.strftime "%Y%m%d_%H%M")
+  )
+    inherited = true if project_id || user_id
+    users = if user_id.nil?
```

Comment:
looks like it's finding deleted users when giving id and not deleted ... how about:

```Ruby
users = (deleted ? User.unscoped : User)
users = users.order(:id)
users = users.where(id: user_id) if user_id
```